### ISSUE TITLE
Overlapping Legend Items on Local Export

### ISSUE THREAD

**Issue by dscrimshaw** [highcharts/highcharts#14044]
#### Expected behaviour
Legend spacing matches that of the chart

#### Actual behaviour
Legend items overlap with each other in exported image

Despite copying the chart options from our export call (with the exception of the export buttons which I had to turn on for the demo), this doesn't quite match my experience in our dev environment, but does still show overlapping legend items. In our case, the png, jpeg and svgs all export fine:
![Trend Analysis (5)](https://user-images.githubusercontent.com/51974703/90225655-93228e80-ddc6-11ea-8001-077b6a0b6bb6.png)

However the last line of the legend on the pdf export has the legend items overlapping and its font very spaced out:
[Trend Analysis (53).pdf](https://github.com/highcharts/highcharts/files/5073228/Trend.Analysis.53.pdf)

What the jsfiddle produces below renders a bit differently but still seems erroneous.

#### Live demo with steps to reproduce
https://jsfiddle.net/denholms/uk87p69v/43/

Export png and pdf image using the context menu, compare with the chart

#### Product version

Highstock 8.1.2

#### Affected browser(s)
Chrome 84.0.4147.89

**Comment by pawelfus** [comment]
Hi @dscrimshaw 

Thank you for reporting the issue. It seems `Nunito-Regular` font is not installed on our exporting server. You need to either change the font, or [setup your own exporting server](https://www.highcharts.com/docs/export-module/setting-up-the-server). In the jsFiddle demo you can also increase [`legend.itemWidth`](https://api.highcharts.com/highcharts/legend.itemWidth): https://jsfiddle.net/BlackLabel/vfpod6es/1/ - but from your PDF I assume it might now be enough.

**Comment by dscrimshaw** [comment]
@pawelfus Hello,

Thanks for the quick response. Unsupported font makes sense. I tried updating and saw the improvement in the demo fiddle but unfortunately not fixing the situation in our dev environment. We're using the Angular Highcharts wrapper, maybe there's some difference there. I did notice for instance it seems to ignore the exporting.chartOptions settings, I tried changing the chart background to white for export rather than transparent and it didn't work unless I modified the chartOptions object (the chart setting, not the exporting setting) or made a call to chart.update prior to the export. 

Updated exporting.allowHTML and that fixed the PDF! But then png background was black, added to exporting.chartOptions when it was ignoring before as I described above, and now it's picking it up and turning it to white. Not sure why that would be the case, seems totally bizarre to me. If you have any guesses as to what might be going on that would be great, but just happy that solved the overlap issue!! :)

Right, we're doing a local export with fallback to server left as default (enabled). I wasn't sure how to put that in the fiddle so I left it as exporting from the context menu. We were somewhat forced to use the fallback to server because on a large chart it was trying to do a regex match with the dataURI string and was provided a filename when it wasn't expecting it so the call kept failing. KacperMadej described the issue we were having here: https://github.com/highcharts/highcharts/issues/10243 , but allowing fallback helped the export to continue.

Sorry for all the extra info, I've been having lots of issues aha. Even somehow got banned from your forum for a month while trying to register, for some reason was labelled a spammer on my first attempt and got banned on my second. 

Anyway, thanks very much for your help!

**Comment by pawelfus** [comment]
Thanks for all of the details! Answers, one by one:

`chartOptions` option doesn't work: make sure you did non set `exporting` config twice. If thats not the case, then I agree, it's strange. Can you recreate this in jsFiddle? I need to check it. This is you demo with green background: https://jsfiddle.net/BlackLabel/265syhwb/ - looks fine.

Banning from forum: This might happen due to your IP used by spammers. I'm not sure if we can do something about it, but could you reach our support directly via email? My apologies for all of the issues!

And I'm happy to hear the issue is now resolved! :)

**Comment by dscrimshaw** [comment]
I may have been setting the exporting config twice, once in the original chartOptions object and then was trying to update the chartOptions.exporting later on when I knew the chart size. I thought if I explicitly referred to chartOptions.exporting.chartHeight or something it wouldn't overwrite the entire exporting object, that they'd get merged. Is that incorrect? I'm on a bit of a time crunch but if I come by that previous issue and can JSFiddle it easily I will.

Setting the html setting to true started causing an issue with a different chart that has a navigator. It used to throw a Highcharts error #18 when exporting PDF, after setting the HTML setting to true it now does that for every export type every time. I tried removing the setting and it went back to the previous behavior (in other words it's repeatable). I still suspect it may be our config at the time of export, I'm looking into it but thought I should mention it. I wasn't able to reproduce in a JSFiddle, and therefore I think something may be adjusting the chartOptions at runtime that I'm unaware of (not the only one working on the product).

I tried sending an email to the address it directed me to but no response as of yet. No worries not your fault :)

Thanks again for your assistance.

**Comment by dscrimshaw** [comment]
I should ask more specifically, in what way should I be updating the exporting options if I need to adjust at runtime? Keeping in mind we're using the Angular wrapper.

Thanks

**Comment by pawelfus** [comment]
By setting `exporting` twice I mean something as simple as this:
```js
exporting: {
 chartOptions: { ... }
},
plotOptions: { ... },
// other options 
exporting: {
  chartOptions: { ... }
}
```
Setting directly `chartOptions.exporting.chartHeight` is correct.

Regarding error18, it's hard to say without seeing live code. It's strange that setting `allowHTML=true` causes this issue. Maybe you are updating navigator/scrollbar the way described in this bug: #13020 ?

> I should ask more specifically, in what way should I be updating the exporting options if I need to adjust at runtime? Keeping in mind we're using the Angular wrapper.

Are you using our official wrapper? If yes, then you will get the best answer from our angular team (just contact us via email :)). If you are using some 3rd party wrapper, please contact authors of that wrapper.

**Comment by dscrimshaw** [comment]
I see, okay good.

Yeah fair enough, not sure what's going on with the navigator. We're just going to leave the navigator out of our export for now as this whole thing has cost too much time and effort. However on this other chart now without the navigator, I'm seeing this overlap on the legend come back, even with the HTML setting as true and the font set to the Arial font family. Any insight?

https://jsfiddle.net/denholms/tcr3vzwd/31/

Yes we're using the official wrapper. Thanks.

**Comment by sebastianbochan** [comment]
The reason is the same like earlier: missing font in our exporting server. When I disabled the custom font (and width) the chart is exported properly:
- https://jsfiddle.net/8xo5ytq2/1/

**Comment by dscrimshaw** [comment]
@sebastianbochan my apologies, set it to Arial specifically to avoid being an unsupported font. It's what the other team on our project seems to be using. Seems I have to delete the fontFamily and width settings rather than set them to undefined, but after that it works. Sorry for the misunderstanding!

**Comment by dscrimshaw** [comment]
I closed this issue as you've answered the cause. But wanted to mention, it would be nice if there was some warnings in the console for unsupported font or something of the sort, and actually a bit more error tracking in general.

**Comment by sebastianbochan** [comment]
Thank you for the hint. It makes sense!

**Comment by pawelfus** [comment]
>  But wanted to mention, it would be nice if there was some warnings in the console for unsupported font

That's a tricky one. Whatever is sent to our exporting server is either SVG or JSON options. Server generates image and sends it back to the browser. If we decide to throw an error there, then we won't send an image. Most of the fonts, even if not supported, don't cause issues like the one you reported. Charts are almost the same and it's fine. If we start throwing errors we will break many Highcharts implementations which we would like to avoid.

And we can not implement this in the library's core neither, because every user can setup her/his own exporting server with different fonts installed.

Long story short, it's tricky :)

**Comment by dscrimshaw** [comment]
Fair enough - is there at least a list of supported fonts somewhere?

**Comment by pawelfus** [comment]
We don't have such list. The best would be to simply test if specific font is available or setup your own exporting server if font is required but not available.
